### PR TITLE
fix(pvt2): migrations from legacy pivot table error when form_data have pieces of pvt2

### PR DESCRIPTION
**Pull request by Always-prog**
### SUMMARY
Hello! 
Migration from legacy pivot table v2 sometimes returns an error due to pieces of  Pivot Table V2 form_data in the legacy pivot. This PR fixes it. Pieces of form_data can occur when users switch from pvt2 tables to legacy pivot.

### TESTING INSTRUCTIONS
1. Create an save Pivot Table v2 visualization
2. Switch to the legacy pivot table and save
3. Try to migrate and see the error "Dublicate key".

Or just try to migrate form_data that is already generated by me and see the error.
```python
{'adhoc_filters': [], 'aggregateFunction': 'Sum', 'applied_time_extras': {}, 'colOrder': 'key_a_to_z', 'colTotals': True, 'columns': ['date_obr'], 'conditional_formatting': [], 'dashboards': [], 'datasource': '2858__table', 'date_format': 'smart_date', 'granularity_sqla': 'date_', 'groupby': ['Name_gr'], 'groupbyColumns': ['date_obr'], 'groupbyRows': ['id_group', 'TT'], 'label_colors': {}, 'metrics': ['count'], 'metricsLayout': 'COLUMNS', 'number_format': 'SMART_NUMBER', 'order_desc': True, 'pandas_aggfunc': 'Sum', 'pivot_margins': True, 'rowOrder': 'value_z_to_a', 'rowTotals': True, 'row_limit': 1000, 'series_limit_metric': ['count'], 'shared_label_colors': {}, 'slice_id': 16619, 'time_grain_sqla': 'P1D', 'time_range': 'Last week', 'timeseries_limit_metric': 'count', 'valueFormat': ',d', 'viz_type': 'pivot_table_v2', 'extra_filters': [], 'dataMask': {}, 'extraControls': {}}
```
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by michael-s-molina** [comment]
> Pieces of form_data can occur when users switch from pvt2 tables to legacy pivot.

Thanks for the fix @Always-prog! If this behavior happens for Pivot Table, it may also happen for other chart types. Would you mind resolving this in a more global way? You can skip rename keys in `base.py`:

```
def _migrate(self) -> None:
   ...
   rv_data = {}
      for key, value in self.data.items():
         // ADD SOMETHING HERE TO SKIP NEW VERSION KEYS
            
         if key in self.rename_keys and self.rename_keys[key] in rv_data:
            raise ValueError("Duplicate key in target viz")
```

**Comment by Always-prog** [comment]
> > Pieces of form_data can occur when users switch from pvt2 tables to legacy pivot.
> 
> Thanks for the fix @Always-prog! If this behavior happens for Pivot Table, it may also happen for other chart types. Would you mind resolving this in a more global way? You can skip rename keys in `base.py`:
> 
> ```
> def _migrate(self) -> None:
>    ...
>    rv_data = {}
>       for key, value in self.data.items():
>          // ADD SOMETHING HERE TO SKIP NEW VERSION KEYS
>             
>          if key in self.rename_keys and self.rename_keys[key] in rv_data:
>             raise ValueError("Duplicate key in target viz")
> ```

Then, can we just instead of an error "Duplicate key in target viz" just skip that keys?

**Comment by michael-s-molina** [comment]
> Then, can we just instead of an error "Duplicate key in target viz" just skip keys which is in rename dictionary?

No because that check is to prevent that different pieces of information are mapped to the same key which is a development error.

**Comment by Always-prog** [comment]
@michael-s-molina I moved removing rename keys from data to the base.py, but before `for key, value in self.data.items():`. Can you look at the code?

File: superset/migrations/shared/migrate_viz/base.py [hunk 1]
```diff
@@ -64,6 +64,12 @@ def _migrate(self) -> None:
         if "viz_type" in self.data:
             self.data["viz_type"] = self.target_viz_type
 
+        # Sometimes visualizations have same keys in the source form_data and rename_keys
+        # We need to remove them from data to allow the migration to work properly with rename_keys
+        for key in self.rename_keys.values():
```
Comment (michael-s-molina) [inline superset/migrations/shared/migrate_viz/base.py:69]: I'm worried we might have keys that belong to the new version but not a corresponding legacy key. In that case, removing the keys will make us lose information. Taking a Pivot Table as example, we might have `series_limit` (new) but no `row_limit` (legacy). For safety, could only remove keys if its legacy counterpart also exists?

Comment (Always-prog) [reply superset/migrations/shared/migrate_viz/base.py]: Got it! I added removing target key only if source and target rename keys exists in form_data

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thank you for the fix @Always-prog!

**Comment by Always-prog** [comment]
@michael-s-molina Hi! If everything is ok with my PR now, can we merge it?

**Comment by michael-s-molina** [comment]
> @michael-s-molina Hi! If everything is ok with my PR now, can we merge it?

Hey. Could you rebase it to fix the Docker issue? I can't merge it with that CI check failing.

**Comment by Always-prog** [comment]
@michael-s-molina There is no need to rebase or merge (I already tried). I created a branch on the latest version of the code. 
Looks like CI broke accidentally, just needs a rerun.

**Comment by michael-s-molina** [comment]
> @michael-s-molina There is no need to rebase or merge (I already tried). I created a branch on the latest version of the code. Looks like CI broke accidentally, just needs a rerun.

You need to have https://github.com/apache/superset/pull/24731 in your commit history to fix the CI error. That's why a rebase is needed.

**Comment by Always-prog** [comment]
@michael-s-molina Got it, thank you. Fixed!

**Pull request merged**